awspricing/__init__.py

```diff
@@ -28,69 +24,48 @@ def _fetch_offers():
     if offers is not None:
         return offers
 
-    resp = session.get(OFFER_BASE_URL + OFFER_INDEX_ENDPOINT)
-    resp.raise_for_status()
-    offers = resp.json()['offers']
+    for product in DEFAULT_AWS_PRODUCTS:
+        _fetch_offer(product)
 
     maybe_write_to_cache(cache_key, offers)
     return offers
 
 
-def _fetch_available_versions(offers, offer_name):
-    cache_key = 'offers_version_{}'.format(offer_name)
-    offers_version = maybe_read_from_cache(cache_key)
-    if offers_version is not None:
-        return offers_version
+def _get_services():
+    global service_list
+    if not service_list:
+        service_list = all_services_names()
+    return service_list
 
-    offers_version_endpoint = offers[offer_name]['versionIndexUrl']
-    resp = session.get(OFFER_BASE_URL + offers_version_endpoint)
-    resp.raise_for_status()
-    offers_version = resp.json()['versions']
 
-    maybe_write_to_cache(cache_key, offers_version)
-    return offers_version
-
-
-def _get_offers():
-    global _OFFERS
-    if not _OFFERS:
-        _OFFERS = _fetch_offers()
-    return _OFFERS
-
-
-def _fetch_offer(offer_name, version='current'):
-    offers = _get_offers()
-    if offer_name not in offers:
-        raise ValueError('Unknown offer name: {}'.format(offer_name))
+def _fetch_offer(offer_name, version=datetime.datetime.now().strftime(TIME_FORMAT)):
```

Comment:
Use time of offer request as "version" now - will this work okay with caching logic? Saw that it was based on the time the path was last altered. 

There is a pricing version per product available () but seems less relevant and harder to access.

Reply:
you'll want to do `version=None` and then `if version == None: version = datetime.datetime.now...`, as the default arg only be evaluated once.

also, should this time be in UTC or is local intentional?

Reply:
Should be UTC; made it `datetime.utcnow()`. 

Also refactored to use `version=None` and set the version in the function